docs/guides/shuffle-sharding.md

```diff
@@ -0,0 +1,94 @@
+
+The idea is to assign each tenant a shard composed by a subset of the Cortex service instances, aiming to minimize the overlapping instances between two different tenants. Shuffle sharding brings the following **benefits** over the default sharding strategy:
+
+- An outage on some Cortex cluster instances/nodes will only affect a subset of tenants.
+- A misbehaving tenant will affect only its shard instances. Due to the low overlapping of instances between different tenants, it's statistically quite likely that any other tenant will run on different instances or only a subset of instances will match the affected ones.
+
+### Low overlapping instances probability
+
+For example, given a Cortex cluster running **50 ingesters** and assigning **each tenant 4** out of 50 ingesters, shuffling instances between each tenant, we get **230K possible combinations**.
+
+Randomly picking two different tenants we have the:
+
+- 71% of chances they will not share any instance
+- 26% of chances they will share only 1 instance
+- 2.7% of chances they will share 2 instances
+- 0.08% of chances they will share 3 instances
+- Only the 0.0004% of chances their instances will fully overlap
+
+![Shuffle sharding probability](/images/guides/shuffle-sharding-probability.png)
+<!-- Chart source at https://docs.google.com/spreadsheets/d/1FXbiWTXi6bdERtamH-IfmpgFq1fNL4GP_KX_yJvbRi4/edit -->
+
+## Cortex shuffle sharding
+
+Cortex currently supports shuffle sharding in the following services:
+
+- [Ingesters](#ingesters-shuffle-sharding)
+- [Query-frontend](#query-frontend-shuffle-sharding)
+- [Store-gateway](#store-gateway-shuffle-sharding)
+
+Shuffle sharding is **disabled by default** and needs to be explicitly enabled in the configuration.
+
+### Guaranteed properties
+
+The Cortex shuffle sharding implementation guarantees the following properties:
+
+- **Stability**<br />
+  Given a consistent state of the hash ring, the shuffle sharding algorithm always selects the same instances for a given tenant, even across different machines.
+- **Consistency**<br />
+  Adding or removing 1 instance from the hash ring leads to only 1 instance changed at most, in each tenant's shard.
+- **Shuffling**<br />
+  Probabilistically and for a large enough cluster, it ensures every tenant gets a different set of instances, with a reduced number of overlapping instances between two tenants to improve failure isolation.
+- **Zone-awareness**<br />
+  When [zone-aware replication](./zone-replication.md) is enabled, the subset of instances selected for each tenant contains a balanced number of instances for each availability zone.
```

Comment:
Should we mention that zone-aware shuffle shard *will* lead to unbalanced distribution of workload between zones, if they have widely different set of instances?

Reply:
It's not just with shuffle sharding enabled. The same happens with shuffle sharding disabled, because with the replication factor is > 1 we'll pick 1 instance per zone. I would document it in the zone-awareness doc (in a separate PR, to unlock this PR).